File: app/src/main/java/acm/androidfitnesseexample/MessagePresenter.java
package acm.androidfitnesseexample;

/**
 * Created in February.
 */
public class MessagePresenter {

    private MessageInterface mMessageInterface;
    private String mText;

    public MessagePresenter(MessageInterface messageInterface) {

        mMessageInterface = messageInterface;
    }

    public void read(String text) {
        mText = text;
    }

    public void produceAnswer() {
        mMessageInterface.showMessage(mText);
    }
}

File: app/src/main/java/acm/androidfitnesseexample/MessagePresenter.kt
package acm.androidfitnesseexample

class MessagePresenter(private val mMessageInterface: MessageInterface) {
    private var mText: String? = null
    fun read(text: String?) {
        mText = text
    }

    fun produceAnswer() {
        mText?.let {
            mMessageInterface.showMessage(it)
        }
    }

}